docs/root/intro/arch_overview/upstream/outlier.rst

```diff
@@ -204,7 +204,9 @@ For gRPC requests, the outlier detection will use the HTTP status mapped from th
 .. note::
 
   If :ref:`active health checking <arch_overview_health_checking>` is also configured, a successful active health check unejects the host and
-  clears all outlier detection counters. If your active health check is not validating data plane traffic then in situations where
+  clears all outlier detection counters. Before the FAILED_ACTIVE_HC health flag is set for the host: a single successful health check unejects.
+  If FAILED_ACTIVE_HC health flag is set for the host: :ref:`healthy_threshold<envoy_v3_api_field_config.core.v3.HealthCheck.healthy_threshold>` long consecutive success health check series unejects (same what also clears FAILED_ACTIVE_HC flag).
+  If your active health check is not validating data plane traffic then in situations where
```

Comment:
If you'll allow me, I think this reads a little better:

```suggestion
  clears all outlier detection counters. If the host is failing health checks, but has not reached the
  :ref:`unhealthy_threshold<envoy_v3_api_field_config.core.v3.HealthCheck.unhealthy_threshold>`, a single successful health check
  will uneject the host. If the FAILED_ACTIVE_HC health flag is set for the host,
  :ref:`healthy_threshold<envoy_v3_api_field_config.core.v3.HealthCheck.healthy_threshold>` consecutive successful health checks
  will uneject the host (and clear the FAILED_ACTIVE_HC flag).
  If your active health check is not validating data plane traffic then in situations where
```

Reply:
The first is part little confusing in this way.  Previous failed checks are not condition.
How about: 'If the host has not reached ... failing health checks, a single succesful ...' ?

Reply:
Sure.